### DISCUSSION TITLE
bug:  WARNING prettier unavailable: Condition failed

### DISCUSSION THREAD

**Discussion by ghost** [LazyVim/LazyVim#4068] [Q&A]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

v0.10.0

### Operating system/version

Ubuntu Mantic

### Describe the bug

Even though i have prettier available, and installed using volta, with volta install prettier, conform.nvim health check says:

conform: require("conform.health").check()

`WARNING prettier unavailable: Condition failed`

### Steps To Reproduce

1. Lazy Sync
2. LazyHealth

### Expected Behavior

It should detect prettier in health checks.

### Repro

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- add any other plugins here
  },
})
```

**Comment by ghost** [comment]
Prettier config check is broken

https://github.com/stevearc/conform.nvim/issues/490#issuecomment-2231922716

**Reply by LevelbossMike** [reply]
For me, the issue was caused by the file type I was working on not being included in lazyvim's `supported` file types. I fixed this by allowing lazyVim's prettier config to be extended with additional file types.

I added the following to https://github.com/LazyVim/LazyVim/blob/150523b77b6e848c4135a97a5fd8f6f79a6f4443/lua/lazyvim/plugins/extras/formatting/prettier.lua#L11

```lua
local additional_supported_filetypes = vim.g.prettier_additional_supported_filetypes or {}
vim.list_extend(supported, additional_supported_filetypes)
```

With that, I can then add additional file types that prettier works on by defining `vim.g.prettier_additional_supported_filetypes`

E.g. in my case:

```lua
-- for adding support for glimmer files in prettier
g.prettier_additional_supported_filetypes = { "javascript.glimmer", "typescript.glimmer" }
```

I am not sure if that aligns with lazyvim's philosophy or not, but for the time being, this is good enough for me.

**Reply by LevelbossMike** [reply]
Coming back to this with some more information:

The config isn't broken. When you set up the parser correctly for the file type you want to edit, lazyvim does the right thing.

This is fairly straightforward to debug if you run the prettier check that lazyvim does manually in the buffer that isn't working as expected:

`prettier --file-info %`

If the parser can't be inferred, you have to fix your prettier-config so that the parser can be inferred. Then it works as expected.

**Reply by deekshithanand** [reply]
@LevelbossMike , thank for looking into this

> The config isn't broken. When you set up the parser correctly for the file type you want to edit, lazyvim does the right thing.

Hey, I am facing the same issue, and pretty new to neovim, I installed the typescript support via lazyExtras, 

what exactly should be done to fix this?

Although format on save works as anticiapted, i did find this error from LazyHealthCheck.

Thanks!

**Reply by LevelbossMike** [reply]
@deekshithanand I edited my prettierrc file and explicitly defined the parser to use for a certian filetype.

In my example:

```
  overrides: [
    {
      files: "*.{gts,gjs}",
      options: {
        parser: "ember-template-tag",
      },
    },

```

**Comment by juridiener** [comment]
Hello i have the same warning in the checkhealth. But in one project it works in a another project it does not. but when i run in both projects the checkhealth i get the same warning. The thing is in the project it didn't work, i get an warning popup when i save the file.

Formatter 'prettier' timeout
And it takes some seconds to save the file. this is more than annoying, not only the formatter does not work in addition it slows me down.

in both prejects i have the same version installed and the same config
